### ISSUE TITLE
Exception: Class auth does not exist

### ISSUE THREAD

**Issue by voidman** [barryvdh/laravel-ide-helper#26]
composer.json:

```
"barryvdh/laravel-ide-helper": "1.*"
```

and got this message below:

```
➜  src  php artisan ide-helper:generate
Exception: Class auth does not exist
Skipping Illuminate\Support\Facades\Auth.
Exception: Class auth does not exist
Skipping Illuminate\Support\Facades\Password.
A new helper file was written to _ide_helper.php
```

**Comment by shehi** [comment]
@barryvdh : I have this same issue after upgrading to L5.1... Any ideas?

**Comment by barryvdh** [comment]
Yes, see above

**Comment by shehi** [comment]
Uhm... Sorry, what you mean? All the posts above are related to L4.1. In L5.1 we have only one config file and mine is similar to official file, nothing wrong there. Am I missing something?

**Comment by barryvdh** [comment]
Then please create a new issue.

**Comment by marre421** [comment]
Had the same issue. Only appeared when i ran composer install --no-dev in prod env. It was my mistake i did not read this:

"instead of adding the service provider in the config/app.php file, you can add the following code to your app/Providers/AppServiceProvider.php file, within the register() method:"

I had Barryvdh\LaravelIdeHelper\IdeHelperServiceProvider::class, in app config and in AppServiceProvider. Removed it and all my problems where solved. Also make sure if you run --no-dev your env should be in production. Hope this helps someone.

**Comment by Aderemi** [comment]
Most of the times the issue of 'class ****(in lower case) does not exist' are caused by something different entirely. If you open bootstrap/cache/services.php you will see the **** that is being complained about.
```   
...
...
     'hash' => 'Illuminate\\Hashing\\HashServiceProvider',
...
...
```
In my own case but Laravel complained ``` class hash does not exist```. With this, I deduced that the problem was not hash, it was something that was thrown before services.php file could be loaded. 
It turned out that I was right I simply dumped ```$e``` on the first line of ```report()``` function in `app/Exceptions/Handler.php` to see the real error which happened to be database connection issues in my case
